uproot4/source/http.py

```diff
@@ -71,32 +71,51 @@ def make_connection(parsed_url, timeout):
         )
 
 
-def get_num_bytes(file_path, parsed_url, timeout):
+def get_response(connection, location):
     """
     Args:
-        file_path (str): The URL to access as a raw string.
-        parsed_url (``urllib.parse.ParseResult``): The URL to access.
-        timeout (None or float): An optional timeout in seconds.
+        connection (``http.client.HTTPConnection``): The connection object from ``make_connection``
+        location (str): The location of the current request, used for error messages
 
-    Returns the number of bytes in the file by making a HEAD request.
+    Gets the response from ``connection.getresponse()`` and validates its status.
     """
-    connection = make_connection(parsed_url, timeout)
-    connection.request("HEAD", parsed_url.path)
     response = connection.getresponse()
 
     if response.status == 404:
         connection.close()
-        raise uproot4._util._file_not_found(file_path, "HTTP(S) returned 404")
+        raise uproot4._util._file_not_found(location, "HTTP(S) returned 404")
+
+    elif str(response.status)[0] == "3":
+        connection.close()
+        raise NotImplementedError(
+            "HTTP(S) redirects are not currently supported "
+            "(https://github.com/scikit-hep/uproot4/issues/121)"
+        )
 
-    if response.status != 200:
+    elif str(response.status)[0] != "2":
         connection.close()
         raise OSError(
-            """HTTP response was {0}, rather than 200, in attempt to get file size
-in file {1}""".format(
-                response.status, file_path
-            )
+            "HTTP response was {}, rather than 2xx, ".format(response.status)
+            + "in attempt to get file size in file {}".format(location)
         )
 
```

Comment:
I'll be sure to apply this policy universally: all the HTTP codes in the 200's are good, all others are bad. But I'll be doing it by checking the integer against the 200's range, not by checking that the first character in the string is "2". I don't know if it's possible to get an error code like 20 or 2000, but it seems like better practice to protect against that.